src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/PartitionedRateLimiter.T.cs

```diff
@@ -119,5 +119,77 @@ public async ValueTask DisposeAsync()
             // Suppress finalization.
             GC.SuppressFinalize(this);
         }
+
+        /// <summary>
+        /// Translates PartitionedRateLimiter&lt;TOuter&gt; into the current <see cref="PartitionedRateLimiter{TResource}"/>
+        /// using the <paramref name="keyAdapter"/> to translate <typeparamref name="TOuter"/> to <typeparamref name="TResource"/>.
+        /// </summary>
+        /// <typeparam name="TOuter">The type to translate into <typeparamref name="TResource"/>.</typeparam>
+        /// <param name="keyAdapter">The function to be called every time a <typeparamref name="TOuter"/> is passed to
+        /// PartitionedRateLimiter&lt;TOuter&gt;.Acquire(TOuter, int) or PartitionedRateLimiter&lt;TOuter&gt;.WaitAsync(TOuter, int, CancellationToken).</param>
+        /// <returns>A new PartitionedRateLimiter&lt;TOuter&gt; that translates <typeparamref name="TOuter"/>
+        /// to <typeparamref name="TResource"/> and calls the inner <see cref="PartitionedRateLimiter{TResource}"/>.</returns>
+        public PartitionedRateLimiter<TOuter> TranslateKey<TOuter>(Func<TOuter, TResource> keyAdapter)
+        {
+            // REVIEW: Do we want to have an option to dispose the inner limiter?
+            // Should the default be to dispose the inner limiter and have an option to not dispose it?
+            // See Stream wrappers like SslStream for prior-art
+            return new TranslatingLimiter<TResource, TOuter>(this, keyAdapter);
```

Comment:
Good point. `StreamWriter` does the same thing with `leaveOpen`, so there's definitely precedent here. I think this is the more convenient behavior considering how this API is likely to be used. Keeping around the original `PartitionedRateLimiter` just to track it for disposal is annoying. Do we like just adding a `bool` parameter?

```c#
public PartitionedRateLimiter<TOuter> TranslateKey<TOuter>(Func<TOuter, TResource> keyAdapter, bool leaveOpen = false);
```

We could add an options type if we think we're going to make this even more configurable in the future, but I think that's probably excessive.

For now, without a `leaveOpen` option, I think we have to default to not taking ownership like you have since that's more flexible although it's less convenient.

Reply:
> Do we like just adding a `bool` parameter?

I think so. I am unable to think of any other potential future options at this time. The prior-art with Stream wrappers and StreamWriter will be helpful when advocating for an API change.